specs/registry-auth.md

```diff
@@ -0,0 +1,143 @@
+
+Alternatively, clients can be authenticated via mutual TLS authentication (mTLS). In other words, clients connect to servers via HTTP over mTLS by presenting client certificates. In this case, authorization can be applied later without further authentication schemes.
+
+
+## Authentication Schemes
+
+Notation supports [basic HTTP authentication][RFC7617] scheme and token schemes including [OAuth 2.0][RFC6749] and [Docker Registry V2 Token][token]. Clients MUST connect to the remote servers via HTTPS if any authentication scheme is applied.
+
+### General Flow
+
+```mermaid
+flowchart LR;
+    ts[Authorization Service];
+    client[Notation];
+    reg[Registry];
+
+    client -- 1. Access resource   --> reg;
+       reg -- 2. Challenge         --> client;
+    client -- 3. Request for token --> ts;
+        ts -- 4. Grant token       --> client;
+    client -- 5. Access with token --> reg;
+       reg -- 6. Response          --> client;
+```
+
+In general, notation clients accesses registry resources as the workflow below, which follows the workflow for [Docker Registry v2 authentication via central service](https://docs.docker.com/registry/spec/auth/token/).
+
+1. Notation attempts to access the remote registry directly. If there is no authentication scheme associated with the registry, skip to *step 6*.
+2. The remote registry returns `401 Unauthorized` with a [WWW-Authenticate](https://datatracker.ietf.org/doc/html/rfc7235#section-4.1) challenge, indicating the required authentication scheme.
+3. Notation requests the authorization service for a bearer token for accessing the target resource with local credentials. If the remote registry requires *basic* scheme, skip to *step 5*.
+4. The authorization grants and returns a bearer token for access back to the notation client.
+5. Notation attempts to access the remote registry again with the obtained bearer token (or local credential for *basic* scheme) in the [Authorization](https://datatracker.ietf.org/doc/html/rfc7235#section-4.2) header.
+6. The remote registry performs the requested operation and returns the response.
+
+Optimization might be performed to reduced the number of requests in order to reduce the overall latency for subsequent requests.
+
+### Basic Scheme
+
+Notation follows [RFC 7617][RFC7617] for the *Basic* HTTP authentication scheme.
+
+If a remote registry is known to support `Basic` authentication scheme, the attempt-challenge phase (*steps 1-4*) can be skipped, and thus Notation can access the remote registry without overhead in terms of the number of requests.
+
+```mermaid
+flowchart LR;
+    client[Notation];
+    reg[Registry];
+
+    client -- Access with credentials --> reg;
+       reg -- Response                --> client;
+```
+
+### Token Scheme
+
+Notation supports two types of token schemes, [Docker][token] and [OAuth 2.0][RFC6749] where OAuth 2.0 is preferred by default.

```

Comment:
What does "preferred by default" mean here? Won't registries typically support one or the other?

I'm trying to understand what a client preference translates to in terms of implementation choices.